pkg/matcher/utils.go

```diff
@@ -33,273 +29,209 @@ type ValidatedJSON struct {
-					}
-					logDiffs.PushBodyDiff(fmt.Sprint(op.OldValue), fmt.Sprint(op.Value), bodyNoise)
-				}
-			} else {
-				logDiffs.PushBodyDiff(fmt.Sprint(tc.HTTPResp.Body), fmt.Sprint(actualResponse.Body), bodyNoise)
-			}
+	case geko.ObjectItems:
+		keys := v2.Keys()
+		vals := v2.Values()
+		for i := range keys {
+			compareSecondResponse(val1, &vals[i], key1, keys[i])
 		}
-		_, err := newLogger.Printf(logs)
-		if err != nil {
-			utils.LogError(logger, err, "failed to print the logs")
+	case map[string]interface{}:
+		for key, val := range v2 {
+			compareSecondResponse(val1, &val, key1, key)
 		}
-
-		err = logDiffs.Render()
-		if err != nil {
-			utils.LogError(logger, err, "failed to render the diffs")
+	case string:
+		if *val1 != v2 {
+			// Reverse the templatized values map.
+			revMap := reverseMap(utils.TemplatizedValues)
+			if _, ok := revMap[*val1]; ok && key1 == key2 {
+				key := revMap[*val1]
+				utils.TemplatizedValues[key] = v2
+				*val1 = v2
+			}
 		}
-	} else {
-		newLogger := pp.New()
-		newLogger.WithLineInfo = false
-		newLogger.SetColorScheme(models.GetPassingColorScheme())
-		var log2 = ""
-		log2 += newLogger.Sprintf("Testrun passed for testcase with id: %s\n\n--------------------------------------------------------------------\n\n", tc.Name)
-		_, err := newLogger.Printf(log2)
-		if err != nil {
-			utils.LogError(logger, err, "failed to print the logs")
+	case float64, int64, int, float32:
+		if *val1 != toString(v2) && key1 == key2 {
+			revMap := reverseMap(utils.TemplatizedValues)
+			if _, ok := revMap[*val1]; ok {
+				key := revMap[*val1]
+				utils.TemplatizedValues[key] = v2
+				*val1 = toString(v2)
+			}
 		}
 	}
-	return pass, res
 }
-
-func FlattenHTTPResponse(h http.Header, body string) (map[string][]string, error) {
-	m := map[string][]string{}
-	for k, v := range h {
-		m["header."+k] = []string{strings.Join(v, "")}
-	}
-	err := AddHTTPBodyToMap(body, m)
-	if err != nil {
-		return m, err
+func reverseMap(m map[string]interface{}) map[interface{}]string {
+	var reverseMap = make(map[interface{}]string)
+	for key, val := range m {
+		reverseMap[val] = key
 	}
-	return m, nil
+	return reverseMap
 }
 
-// UnmarshallJSON returns unmarshalled JSON object.
-func UnmarshallJSON(s string, log *zap.Logger) (interface{}, error) {
-	var result interface{}
-	if s == "" {
-		return nil, nil
-	}
-	if err := json.Unmarshal([]byte(s), &result); err != nil {
-		utils.LogError(log, err, "cannot convert json string into json object", zap.String("json", s))
-		return nil, err
-	}
-	return result, nil
-}
-
-func ArrayToMap(arr []string) map[string]bool {
-	res := map[string]bool{}
-	for i := range arr {
-		res[arr[i]] = true
-	}
-	return res
-}
-
-func InterfaceToString(val interface{}) string {
+// Remove all types of value to strings for comparison.
+func toString(val interface{}) string {
 	switch v := val.(type) {
 	case int:
-		return fmt.Sprintf("%d", v)
+		return strconv.Itoa(v)
```

Comment:
Don't change the current implementation if you don't require it, otherwise we have to test it for multiple cases.